DFImageManager/Source/Core/Managing/DFImageManager.m

```diff
@@ -245,6 +245,11 @@ - (BOOL)canHandleRequest:(nonnull DFImageRequest *)request {
     return [_conf.fetcher canHandleRequest:request];
 }
 
+- (BOOL)hasCacheWithRequest:(nonnull DFImageRequest *)request {
+    NSParameterAssert(request);
+    return [self _cachedResponseForRequest:request];
```

Comment:
Unlike `-[SDWebImageManager.sharedManager cachedImageExistsForURL:imageURL]` the `[self _cachedResponseForRequest:request]` would only return YES when the image is in top level memory cache.

There are two cache "layers":
- The `DFImageManager` class provides a top level memory cache for processed (decompressed/transformed etc) images (see `DFImageCaching` protocol and `DFImageCache` class). `[self _cachedResponseForRequest:request]` method only checks if the image is stored in this memory cache. [More info](https://github.com/kean/DFImageManager/wiki/Advanced-Image-Caching-Guide)
- The class that conforms to `DFImageFetching` protocol might provide it's own memory and/or persistent cache. For instance `DFURLImageFetcher <DFImageFetching>` uses `NSURLCache` for that purposes. [More info](https://github.com/kean/DFImageManager/wiki/Image-Caching-Guide). And in some cases image are simply stored locally (see `DFAssetsLibraryImageFetcher`), there is no "cache".

Btw, is there really a need for `-hasCacheWithRequest:`? "Cache" is a transient thing, this method might return YES, but the image would not be available seconds later.